### ISSUE TITLE
ERROR: Unexpected segmentation fault encountered in worker.

### ISSUE THREAD

**Issue by brtknr** [fastai/fastai#826]
With `fastai v1`, `pytorch-nightly` on `macOS 10.13`, I can run the following without any issues:

```
%reload_ext autoreload
%autoreload 2

from fastai import *
from fastai.vision import *
from fastai.docs import *

untar_data(MNIST_PATH)
data = image_data_from_folder(MNIST_PATH)
learn = ConvLearner(data, tvm.resnet18, metrics=accuracy)
learn.fit(1)
```

However, if I try to run the example given in `./examples/dogs_cats.ipynb`, the python kernel topples with the following error:

```
~/Envs/ML/lib/python3.6/site-packages/torch/utils/data/dataloader.py in handler(signum, frame)
    236         # This following call uses `waitid` with WNOHANG from C side. Therefore,
    237         # Python can still get and update the process status successfully.
--> 238         _error_if_any_worker_fails()
    239         if previous_handler is not None:
    240             previous_handler(signum, frame)

RuntimeError: DataLoader worker (pid 55095) is killed by signal: Unknown signal: 0. Details are lost due to multiprocessing. Rerunning with num_workers=0 may give better error trace.
```

When I set `num_workers=0` inside `image_data_from_folder` function, the cell just freezes. Has anyone had any luck with this?

Looking a the Jupyter console log, I see the error given in the title. I am not sure how to debug this.

**Comment by jph00** [comment]
Sorry we don't support fastai on Mac. If you find and fix issues we're happy to merge a PR, however.

**Comment by brtknr** [comment]
I got fastai v1 working on my Mac by following the instructions on pytorch/pytorch to install pytorch from source.

I managed to also address this issue by reducing the image size from 228 to 60 pixels :( ... I might just have been running out of memory.

**Comment by stocks29** [comment]
I was running into this same issue on my mac. Ran fine with `num_workers=0` but not with more than zero unless I set the pixels to 80 or lower.

I made a few changes and now things are working:
- switched from python 3.6.8 to 3.7.6
- increased shared memory settings as described here: http://flummox-engineering.blogspot.com/2014/05/increasing-shared-memory-for-os-x.html
- installed latest versions of all libs as of this writing installed from pip

Unfortunately I did all of these things at once so it's unclear which of them actually solved the issue for me. If I have some time later, I may try to pinpoint but in a bit of a rush right now.

I'm now able to train with `num_workers=16` and 500 pixel images without issue. I hope this helps someone.
